src/app/editor.js

```diff
@@ -19,8 +19,9 @@ function Editor (loadingFromGist) {
   this.uploadFile = function (file, callback) {
     var fileReader = new FileReader();
 
-    SOL_CACHE_FILE = utils.fileKey(file.name);
+    var cacheName = utils.fileKey(file.name);
     fileReader.onload = function (e) {
+      SOL_CACHE_FILE = cacheName;
```

Comment:
Is the browser stalled while the file loads? Is this safe if the user switches the current file in the meantime?